WalletWasabi/Backend/Models/FilterModel.cs

```diff
@@ -31,6 +30,12 @@ public FilterModel(SmartHeader header, Lazy<GolombRiceFilter> filter)
 	// is constructed.This ensures the key is deterministic while still varying from block to block.
 	public byte[] FilterKey => Header.BlockHash.ToBytes()[..16];
 
+	public static FilterModel FromParameters(uint blockHeight, uint256 blockHash, byte[] filterData, uint256 prevBlockHash, long blockTime)
+	{
+		Lazy<GolombRiceFilter> filter = new(() => new GolombRiceFilter(filterData, 20, 1 << 20), LazyThreadSafetyMode.ExecutionAndPublication);
```

Comment:
I know this is not new in this PR but I think it worth to mention. The lazy loading here is an important optimization because the headers chain and the filters share the same data structure when they should not. The idea of having a headers chain is similar to blockheaders chain and block: make it possible to verify the headers chain and once we verify it is correct then we can download the blocks. The same is true here, you should be able to verify that the headers are okay without needing to download the filters.